metrics/tagged.go

```diff
@@ -0,0 +1,37 @@
+package metrics
+
+// taggedMetrics is a metrics publisher decorator that add predefined tags to all metrics
+type taggedMetrics struct {
+	Metrics
+	tags []Tag
+}
+
+// NewTaggedMetrics returns a new metrics publisher with predefined tags for all the metrics
+func NewTaggedMetrics(m Metrics, tags ...Tag) Metrics {
+	return &taggedMetrics{Metrics: m, tags: tags}
+}
+
+// Provide a Counter with the given name and tags
+func (m *taggedMetrics) Counter(name string, tags ...Tag) Counter {
+	return m.Metrics.Counter(name, m.tags...).WithTags(tags...)
```

Comment:
Maybe it's out of the scope of what you want to achieve in this PR.
But this is confusing, I would expect tags to be overwritten, it would be more clear to do merging here or change the WithTags function to be called AddTags.
Feel free to ignore me tough :P

Reply:
I think that WithTags will override the predefined tags, I'll check it

Reply:
@sp-daniel-ramos I tested and doesn't overwrites the tags, just adds a new one with the same key.

We can rename WithTags to AddTags if everyone agree (in other pr 😄)

Reply:
Yes, it's confusing I would do it in a separate PR.